### PR TITLE
add scoring for QDM Measure at measure level

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR Template

Jira Ticket: [MAT-5533](https://jira.cms.gov/browse/MAT-5533)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e. Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/models/validators/QDMMeasureScoringValidator.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+				MeasureScoring scoring = MeasureScoring.valueOfText(measure.getScoring());
+				if(scoring==null) {
+					return false;
+				}
+			}
+		} 
+		else {
+			return true;
+		}
+		return true;
```
Comment (RohitKandimalla) [inline src/main/java/gov/cms/madie/models/validators/QDMMeasureScoringValidator.java:32]: Looks like one of the return true statements can be avoided.

File: src/main/java/gov/cms/madie/models/measure/Measure.java [hunk 2]
```diff
@@ -129,6 +131,8 @@ public class Measure extends ResourceAcl {
   private String cmsId;
   
   private ReviewMetaData reviewMetaData = new ReviewMetaData();
+  
+  private String scoring;
```
Comment (RohitKandimalla) [inline src/main/java/gov/cms/madie/models/measure/Measure.java:135]: Does the Scoring field in groups also requires an update to its validation ? Currently, Group.scoring is set to be not null, but for QDM measures that field can be NULL.

Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/models/measure/Measure.java:135]: @RohitKandimalla The Measure level scoring is required only for QDM Measures. As for Group.scoring validation, we could either set it to the Measure.scoring for QDM Measures, or using a different model. So at this point, it's probably ok to leave Group.scoring as is. Thanks for the suggestion though.

File: src/main/java/gov/cms/madie/models/validators/QDMMeasureScoringValidator.java (hunk 1 above)
Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/models/validators/QDMMeasureScoringValidator.java:29]: @RohitKandimalla Thanks Rohit, I just removed the unnecessary return, and also update java version to be 17.

**Comment by sb-cecilialiu** [comment]
Closing this PR as we have new model change, replaced by this PR: https://github.com/MeasureAuthoringTool/madie-java-models/pull/122

**Pull request closed without merging**